x-pack/plugins/monitoring/server/lib/cluster/get_index_patterns.test.ts

```diff
@@ -14,89 +15,182 @@ const getConfigWithCcs = (ccsEnabled: boolean) => {
+    (type, expected) => {
+      const indexPatterns = getNewIndexPatterns({
+        type,
+        config: getConfigWithCcs(true),
+        moduleType: 'kibana',
+      });
+      expect(indexPatterns).toBe(expected);
+    }
+  );
+  it.each<TestTuple>([
+    [undefined, '.monitoring-logstash-*,metrics-logstash.*-*'],
+    ['metrics', '.monitoring-logstash-*,metrics-logstash.*-*'],
+    ['logs', 'filebeat-*,logs-logstash.*-*'],
+  ])(
+    'returns logstash index patterns when ccs is enabled and no ccs payload and type %s',
+    (type, expected) => {
+      const indexPatterns = getNewIndexPatterns({
+        type,
+        config: getConfigWithCcs(true),
+        moduleType: 'logstash',
+      });
+      expect(indexPatterns).toBe(expected);
+    }
+  );
+  it.each<TestTuple>([
+    [undefined, '.monitoring-beats-*,metrics-beats.*-*'],
+    ['metrics', '.monitoring-beats-*,metrics-beats.*-*'],
+    ['logs', 'filebeat-*,logs-beats.*-*'],
+  ])(
+    'returns beats index patterns when ccs is enabled and no ccs payload and type %s',
+    (type, expected) => {
+      const indexPatterns = getNewIndexPatterns({
+        type,
+        config: getConfigWithCcs(true),
+        moduleType: 'beats',
+      });
+      expect(indexPatterns).toBe(expected);
+    }
+  );
+  it.each<TestTuple>([
+    [undefined, '.monitoring-es-*,metrics-elasticsearch.cluster_stats-*'],
+    ['metrics', '.monitoring-es-*,metrics-elasticsearch.cluster_stats-*'],
+    ['logs', 'filebeat-*,logs-elasticsearch.cluster_stats-*'],
+  ])('returns elasticsearch index patterns with dataset and type %s', (type, expected) => {
     const indexPatterns = getNewIndexPatterns({
+      type,
       config: getConfigWithCcs(true),
       moduleType: 'elasticsearch',
       dataset: 'cluster_stats',
     });
-    expect(indexPatterns).toBe('.monitoring-es-*,metrics-elasticsearch.cluster_stats-*');
-  });
-  it('returns elasticsearch index patterns without ccs prefixes when ccs is disabled', () => {
-    const indexPatterns = getNewIndexPatterns({
-      config: getConfigWithCcs(false),
-      moduleType: 'elasticsearch',
-    });
-    expect(indexPatterns).toBe('.monitoring-es-*,metrics-elasticsearch.*-*');
+    expect(indexPatterns).toBe(expected);
   });
-  it('returns elasticsearch index patterns without ccs prefixes when ccs is disabled but ccs request payload has a value', () => {
+  it.each<TestTuple>([
+    [undefined, '.monitoring-es-*,metrics-elasticsearch.*-*'],
+    ['metrics', '.monitoring-es-*,metrics-elasticsearch.*-*'],
+    ['logs', 'filebeat-*,logs-elasticsearch.*-*'],
+  ])(
+    'returns elasticsearch index patterns without ccs prefixes when ccs is disabled and type %s',
+    (type, expected) => {
+      const indexPatterns = getNewIndexPatterns({
+        type,
+        config: getConfigWithCcs(false),
+        moduleType: 'elasticsearch',
+      });
+      expect(indexPatterns).toBe(expected);
```

Comment:
The repetition in these test cases still makes me twitch a little, but maybe best left for future enhancement.

Reply:
fixed in https://github.com/elastic/kibana/pull/139121/commits/7ed3bbfbce5cffafa4e4bdbccaa58e9a7c868dea